src/chainstate/stacks/miner.rs

```diff
@@ -1464,6 +1495,9 @@ impl StacksBlockBuilder {
         let mut epoch_tx = builder.epoch_begin(&mut chainstate, burn_dbconn)?;
         builder.try_mine_tx(&mut epoch_tx, coinbase_tx)?;
 
+        mempool.reset_last_known_nonces()?;
+        mempool.estimate_tx_rates(estimator, metric, 100)?;
```

Comment:
Because there can be far more than 100 transactions available for consideration, what's our strategy for helping miners find the *maximum-fee-rate* transactions, if we're only going to estimate fees for 100 of them per attempt at building an anchored block?

Reply:
The estimates aren't wiped out between blocks --- given the growth rate of the mempool, in practice, the miner would construct estimates for all the transactions in the mempool. Alternatively, the miner could do this iteratively -- if there's more than 100 transactions without estimates in the mempool, the miner could try to estimate more of the transactions after doing one pass of block assembly.